src/main/java/com/denizenscript/clientizen/serverevents/ServerEventManager.java

```diff
@@ -0,0 +1,59 @@
+package com.denizenscript.clientizen.serverevents;
+
+import com.denizenscript.clientizen.network.Channels;
+import com.denizenscript.clientizen.network.DataSerializer;
+import com.denizenscript.clientizen.network.NetworkManager;
+import com.denizenscript.denizencore.utilities.debugging.Debug;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ServerEventManager {
+
+	public static final Map<String, ServerEvent> serverEvents = new HashMap<>();
+
+	public static void init() {
+		NetworkManager.registerInChannel(Channels.EVENT_DATA, message -> {
+			for (ServerEvent event : serverEvents.values()) {
+				if (event.enabled) {
+					event.enabled = false;
+					event.disable();
+				}
+			}
+			int size = message.readInt();
+			for (int i = 0; i < size; i++) {
+				String id = message.readString();
+				ServerEvent event = serverEvents.get(id);
+				if (event == null) {
+					Debug.echoError("Invalid event '" + id + "' received from server!");
```

Comment:
Maybe add a "is your Clientizen version out of date?" to the end

Reply:
Good idea, added